python/src/sdf/pyVisual.cc

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "pyVisual.hh"
+
+#include <pybind11/pybind11.h>
+
+#include "sdf/Visual.hh"
+#include "sdf/Geometry.hh"
+#include "sdf/Material.hh"
+
+using namespace pybind11::literals;
+
+namespace sdf
+{
+// Inline bracket to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+namespace python
+{
+/////////////////////////////////////////////////
+void defineVisual(pybind11::object module)
+{
+  pybind11::class_<sdf::Visual>(module, "Visual")
+    .def(pybind11::init<>())
+    .def(pybind11::init<sdf::Visual>())
+    .def("name", &sdf::Visual::Name,
+         "Get the name of the visual. "
+         "The name of the visual must be unique within the scope of a Link.")
+    .def("set_name", &sdf::Visual::SetName,
+         "Set the name of the visual. "
+         "The name of the visual must be unique within the scope of a Link.")
+    .def("cast_shadows", &sdf::Visual::CastShadows,
+         "Get whether the visual casts shadows")
+    .def("set_cast_shadows", &sdf::Visual::SetCastShadows,
+         "Set whether the visual casts shadows")
+    .def("transparency", &sdf::Visual::Transparency,
+         "Get the transparency value of the visual")
+    .def("set_transparency", &sdf::Visual::SetTransparency,
+         "Set the transparency value for the visual")
+    .def("geometry", &sdf::Visual::Geom,
+         pybind11::return_value_policy::reference,
+         "Get a pointer to the collisions's geometry.")
+    .def("set_geometry", &sdf::Visual::SetGeom,
+         "Set the collision's geometry")
+    .def("raw_pose", &sdf::Visual::RawPose,
```

Comment:
But it's a `const` reference, I thought is fine to return this as a copy. @scpeters ?

Reply:
Most of the items we return are small objects, so returning a copy makes sense. I just wanted to make sure we were doing this intentionally. Technically, it would be a change in behavior in that if I get a raw pose by calling `raw_pose` then call `set_raw_pose`, it wouldn't affect the raw pose I obtained earlier whereas in C++ it would. 

It would be good to document these kind of differences between the C++ and Python API somewhere. Are we generating python docs and uploading them somewhere for ign-math? If we don't have the infrastructure setup yet, maybe we should put a README in the `src/python`.